src/main/java/seedu/address/model/person/Class.java

```diff
@@ -3,42 +3,148 @@
 import java.time.DateTimeException;
 import java.time.LocalDate;
 import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
 
 /**
  * Represents a Person's class date in the address book.
- * Guarantees: immutable.
+ * Guarantees: immutable; is valid as declared in {@link #isValidClassString(String)}
  */
 public class Class {
+
     public static final String MESSAGE_CONSTRAINTS = "Class can take any string"
-            + " in the format of 'yyyy-MM-dd 0000-2300'";
+            + " in the format of 'yyyy-MM-dd 0000-2359'";
     public static final String INVALID_DATETIME_ERROR_MESSAGE = "Date should be a valid date";
     public static final String INVALID_TIME_ERROR_MESSAGE = "Time should be in the range of 0000 - 2359";
+    public static final String INVALID_DURATION_ERROR_MESSAGE = "EndTime must be after StartTime";
     public static final String VALIDATION_DATETIME_REGEX = "[0-9]{4}[-][0-9]{2}[-][0-9]{2}";
     public static final String VALIDATION_TIME_REGEX = "[0-9]{4}";
     public static final String VALIDATION_CLASS_REGEX = VALIDATION_DATETIME_REGEX
             + "[ ]" + VALIDATION_TIME_REGEX + "[-]" + VALIDATION_TIME_REGEX;
+
     public final LocalDate date;
     public final LocalTime startTime;
     public final LocalTime endTime;
 
+    public final String classDateTime; //User input eg. 2022-05-05 1200-1500
+    public final String classToString; //formatted date eg. 5 May 2022 12PM-3PM
+
+    /**
+     * Constructs a {@code Class}.
+     *
+     * @param classDateTime Defaulted to be "".
+     */
+    public Class(String classDateTime) {
+        this.date = null;
+        this.startTime = null;
+        this.endTime = null;
+        this.classDateTime = "";
+        this.classToString = "";
+    }
 
     /**
      * Constructs a {@code Class}.
      *
-     * @param date
-     * @param startTime
-     * @param endTime
+     * @param date LocalDate object.
+     * @param startTime LocalTime object.
+     * @param endTime LocalTime object.
      */
-    public Class(LocalDate date, LocalTime startTime, LocalTime endTime) {
+    public Class(LocalDate date, LocalTime startTime, LocalTime endTime, String classDateTime) {
```

Comment:
I think a more OOP way to store the String representation of the class (ie the field `classToString` in this case) is to write a `toString()` method and call the method when we need it instead of storing an additional field for this purpose.